### PR TITLE
Личный проект: пока все дома

### PR DESCRIPTION
**Pull request by costigans**
---
:mortar_board: [Личный проект: пока все дома](https://up.htmlacademy.ru/javascript/14/user/243814/tasks/13)

:boom: https://htmlacademy-javascript.github.io/243814-keksobooking/2/

### CODE REVIEW

File: js/map.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
+'use strict';
+
+var noticesAmount = 8;
+var OFFER_TITLE = ['Большая уютная квартира', 'Маленькая неуютная квартира', 'Огромный прекрасный дворец', 'Маленький ужасный дворец', 'Красивый гостевой домик', 'Некрасивый негостеприимный домик', 'Уютное бунгало далеко от моря', 'Неуютное бунгало по колено в воде'];
```
Comment (rkhaslarov) [inline js/map.js]: Name of the variable should be in plural.

File: js/map.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+var OFFER_TYPE = ['palace', 'flat', 'house', 'bungalo'];
```
Comment (rkhaslarov) [inline js/map.js]: Name of the variable should be in plural.

File: js/map.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  var newArr = arr.slice();
+  newArr.sort(function () {
+    return Math.random() - 0.5;
+  });
+
+  return newArr;
+};
+
+/**
+ * generate new array with sorting elemeents and random length
```
Comment (rkhaslarov) [inline js/map.js]: generates ... elements

File: js/map.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+ *
+ * @param {Array} arr
+ * @return {Array}
+ */
+var getNewArray = function (arr) {
```
Comment (rkhaslarov) [inline js/map.js]: Rename to getRandomValuesFromArray

File: js/map.js [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  var newArray = arr.slice();
+  var i = getRandomNumberInRange(1, (arr.length + 1));
+  newArray = shuffleArray(newArray);
+  newArray.length = i;
+  return newArray;
+};
+
+/**
+ * generate and returns array of objects-notices
```
Comment (rkhaslarov) [inline js/map.js]: generates

File: js/map.js [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+ *
+ * @param  {Number} amount – array notices
+ * @return {Array}
+ */
+var getNotices = function (amount) {
```
Comment (rkhaslarov) [inline js/map.js]: generateAdsInfo

File: js/map.js [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+  var notices = [];
```
Comment (rkhaslarov) [inline js/map.js]: ads

File: js/map.js [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+        'y': coordinateY,
+      }
+    };
+
+    notices.push(notice);
+  }
+  return notices;
+};
+
+var notices = getNotices(noticesAmount);
```
Comment (rkhaslarov) [inline js/map.js]: ads

File: js/map.js [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+
+/**
+ * clones essential pin template and modifies it
+ *
+ * @param {Object} notice
+ * @return {Object}
+ */
+var generatePinFromNotice = function (notice) {
+  var pin = noticeTemplate.querySelector('.map__pin').cloneNode(true);
```
Comment (rkhaslarov) [inline js/map.js]: Move it up

File: js/map.js [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+
+var notices = getNotices(noticesAmount);
+
+/**
+ * clones essential pin template and modifies it
+ *
+ * @param {Object} notice
+ * @return {Object}
+ */
+var generatePinFromNotice = function (notice) {
```
Comment (rkhaslarov) [inline js/map.js]: renderPin

File: js/map.js [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+  pinImage.src = notice.author.avatar;
+  pinImage.alt = notice.offer.title;
+
+  return pin;
+};
+
+/**
+ * render all pins by notices and returns dom-fragment
+ *
+ * @return {Object} // ???
```
Comment (rkhaslarov) [inline js/map.js]: ???

File: js/map.js [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+
+  pin.style = 'left: ' + pinX + 'px; top: ' + pinY + 'px;';
+  pinImage.src = notice.author.avatar;
+  pinImage.alt = notice.offer.title;
+
+  return pin;
+};
+
+/**
+ * render all pins by notices and returns dom-fragment
```
Comment (rkhaslarov) [inline js/map.js]: renders

File: js/map.js [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+
+mapPins.appendChild(renderAllPins());
+
+/**
+ * translate a value of type to the string equivalent
+ *
+ * @param {Object} type
+ * @return {String}
+ */
+var translateTypeToString = function (type) {
```
Comment (rkhaslarov) [inline js/map.js]: Use object

File: js/map.js (hunk 1 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 2 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 3 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 4 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 5 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 7 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 6 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 8 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 9 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js [hunk 14]
```diff
@@ -0,0 +1,233 @@
+'use strict';
+
+var adsAmount = 8;
```
Comment (rkhaslarov) [inline js/map.js]: should be named as constant

File: js/map.js [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+var OFFERS_TITLES = ['Большая уютная квартира', 'Маленькая неуютная квартира', 'Огромный прекрасный дворец', 'Маленький ужасный дворец', 'Красивый гостевой домик', 'Некрасивый негостеприимный домик', 'Уютное бунгало далеко от моря', 'Неуютное бунгало по колено в воде'];
+var OFFERS_TYPES = ['palace', 'flat', 'house', 'bungalo'];
+var OFFERS_TIME = ['12:00', '13:00', '14:00'];
```
Comment (rkhaslarov) [inline js/map.js]: OFFERS_DATES

File: js/map.js [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+  return arr[Math.floor(Math.random() * arr.length)];
+};
+
+/**
+ * random sorting of array elements
+ *
+ * @param {Array} arr
+ * @return {Array}
+ */
+var shuffleArray = function (arr) { // По идее здесь нужно "изымать" значения не меня порядок
```
Comment (rkhaslarov) [inline js/map.js]: remove unnecessary comment

File: js/map.js [hunk 17]
```diff
@@ -0,0 +1,233 @@
...
+mapPins.appendChild(renderAllPins());
+
+/**
+ * translate a value of type to the string equivalent
+ *
+ * @param {Object} type
+ * @return {String}
+ */
+var translateTypeToString = function (type) {
+  var obj = {
```
Comment (rkhaslarov) [inline js/map.js]: Move it up and name it as constant and remove this unnecessary function

File: js/map.js [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+var MAX_Y = 630;
+
+// Show map
+var map = document.querySelector('.map');
+map.classList.remove('map--faded');
+
+var mapPins = document.querySelector('.map__pins');
+var adTemplate = document.querySelector('template').content;
+var mapFilters = document.querySelector('.map__filters-container');
+var pin = adTemplate.querySelector('.map__pin').cloneNode(true);
```
Comment (rkhaslarov) [inline js/map.js]: remove cloneNode here and add it in renderPinFromAd, please change the name of variable to pinTemplate

File: js/map.js (hunk 13 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 14 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 15 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 16 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 17 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 18 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js [hunk 19]
```diff
@@ -0,0 +1,217 @@
...
+
+/**
+ * generates new array with sorting elemeents and random length
+ *
+ * @param {Array} arr
+ * @return {Array}
+ */
+var getRandomValuesFromArray = function (arr) {
+  var newArray = arr.slice();
+  var i = getRandomNumberInRange(1, (arr.length + 1));
```
Comment (rkhaslarov) [inline js/map.js]: unnecessary variable, please assign expression directly to length property of array

File: js/map.js [hunk 20]
```diff
@@ -0,0 +1,217 @@
...
+        'y': coordinateY,
+      }
+    };
+
+    ads.push(ad);
+  }
+  return ads;
+};
+
+var ads = generateAdsInfo(ADS_AMOUNT);
```
Comment (rkhaslarov) [inline js/map.js]: move it down after all functions declarations, but before rendering

File: js/map.js [hunk 21]
```diff
@@ -0,0 +1,217 @@
...
+ *
+ * @param {Array} photos
+ * @return {String} - html
+ */
+var getCardPhotos = function (photos) {
+  var container = '';
+
+  for (var i = 0; i < photos.length; i++) {
+    container += '<img src="' + photos[i] + '" class="popup__photo" width="45" height="40" alt="Фотография жилья">';
+    // Нужно ли сделать более гибко, через createElement и classList например?
```
Comment (rkhaslarov) [inline js/map.js]: No need, remove the comment

File: js/map.js (hunk 19 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 21 above)
Comment (costigans) [reply js/map.js]: fix

File: js/map.js (hunk 20 above)
Comment (costigans) [reply js/map.js]: fix

**Pull request merged**